src/main.cpp

```diff
@@ -310,11 +322,26 @@ int main(int argc, char * argv[])
     if (!(ErrorDialogHandler::instance()->checkError())) {
         qDebug() << "Displaying mixxx";
         mixxx->show();
+        splash.finish(mixxx);
 
         qDebug() << "Running Mixxx";
         result = a.exec();
     }
 
+    QMessageBox cleaningUpBox;
+    cleaningUpBox.setStandardButtons(0);
+    cleaningUpBox.setWindowTitle(QObject::tr("Exiting Mixxx"));
+    cleaningUpBox.setText(QObject::tr(
+            "Cleaning up the library. This may take some time.\n"
+            "Do not restart Mixxx or shutdown the device until\n "
+            "this box has disappeared."));
```

Comment:
> I am confused, how is it possible to shut down mixxx early?

If you kill it and if you power off your PC. 

Worst Scenario: 
* Mixxx hangs
* Close Mixxx
* Restart Mixxx
* Can't get the Sound hardware because of first Mixxx instance is still cleaning the DB file.   
* Kill the first Mixxx (Because the user did not know the fact) 
* Can't Start because DB is corrupt :-( 

> If Mixxx can't safely be run while another instance of mixxx is running, we should prevent that instead. Relying on the user is not a solution.

We already have a kind of locking via the ALSA devices, which can be used by a single process only. 

I am not sure what can happen if we run two Instance of Mixxx on the same Sqlight file. In a short test it just works, if you start one after the other. However, Mixxx is reporting a probably corrupt DB if you start the second DB during shutdown of the first Mixxx instance.   
In the Sqlite docks, it is suggested to lock the database file. This would prevent all database related race conditions. But even if we do this a window indicating that Mixxx is still alive will help.  
So IMHO this PR is a good step in the right direction. Additional locking can be done in second PR. 

Did you thought about an improved text?